lib/utils/combineJSON.js

```diff
@@ -60,14 +60,19 @@ function combineJSON(arr, deep, collision, source, parsers=[]) {
 
       // Iterate over custom parsers, if the file path matches the parser's
       // pattern regex, use it's parse function to generate the object
-      parsers.forEach(({pattern, parse}) => {
-        if (resolvedPath.match(pattern)) {
-          file_content = parse({
+      // Allows async parsers
+      for (let i = 0; i < parsers.length; i++) {
+        if (resolvedPath.match(parsers[i].pattern)) {
+          let result = parsers[i].parse({
             contents: fs.readFileSync(resolvedPath, {encoding:'UTF-8'}),
             filePath: resolvedPath
           });
+          if (result instanceof Promise) {
```

Comment:
you can await even the result is not a promise IIRC. simplifying this call

Reply:
TIL lol, you're right.